Build failed

```sh
$ make vet
/home/coder/go/bin/go1.19 vet ./...
# github.com/cosmo-workspace/cosmo/internal/dashboard
internal/dashboard/auth_handler.go:84:44: cannot use req (variable of type "github.com/cosmo-workspace/cosmo/api/openapi/dashboard/v1alpha1".LoginRequest) as type *authproxy.LoginRequest in argument to authrizer.Authorize
# github.com/cosmo-workspace/cosmo/internal/dashboard
vet: internal/dashboard/auth_handler.go:84:44: cannot use req (variable of type "github.com/cosmo-workspace/cosmo/api/openapi/dashboard/v1alpha1".LoginRequest) as *authproxy.LoginRequest value in argument to authrizer.Authorize
make: *** [Makefile:143: vet] エラー 1

$ make go-test TEST_FILES=./internal/dashboard/
KUBEBUILDER_ASSETS="/home/coder/.local/share/kubebuilder-envtest/k8s/1.25.0-linux-amd64" \
/home/coder/go/bin/go1.19 test ./internal/dashboard/ -coverprofile cover.out 
# github.com/cosmo-workspace/cosmo/internal/dashboard [github.com/cosmo-workspace/cosmo/internal/dashboard.test]
internal/dashboard/auth_handler.go:84: cannot use req (variable of type "github.com/cosmo-workspace/cosmo/api/openapi/dashboard/v1alpha1".LoginRequest) as type *authproxy.LoginRequest in argument to authrizer.Authorize
FAIL    github.com/cosmo-workspace/cosmo/internal/dashboard [build failed]
FAIL
make: *** [Makefile:167: go-test] エラー 1
```

It is caused by the change of Authorizer interface

```diff
type Authorizer interface {
-	Authorize(ctx context.Context, req dashv1alpha1.LoginRequest) (bool, error)
+	Authorize(ctx context.Context, msg *authv1alpha1.LoginRequest) (bool, error)
}
```

--- a/api/openapi/dashboard/v1alpha1/auth_request_impl.go
+++ b/api/openapi/dashboard/v1alpha1/auth_request_impl.go
@@ -1,0 +1,13 @@
+/*
+ * Temporary patch for https://github.com/cosmo-workspace/cosmo/issues/598
+ */
+
+package v1alpha1
+
+func (r *LoginRequest) GetId() string {
+	return r.Id
+}
+
+func (r *LoginRequest) GetPassword() string {
+	return r.Password
+}

--- a/hack/api-generate/Makefile
+++ b/hack/api-generate/Makefile
@@ -6,6 +6,7 @@ generate: go ts
 
 go: install
 	yarn go:gen_nofix
+	cp go-server/auth_request_impl.go_ ../../api/openapi/dashboard/v1alpha1/auth_request_impl.go
 	$(GOBIN)/goimports -w ../../api/openapi/dashboard/v1alpha1/*.go
 
 ts: install

--- a/hack/api-generate/go-server/auth_request_impl.go_
+++ b/hack/api-generate/go-server/auth_request_impl.go_
@@ -1,0 +1,13 @@
+/*
+ * Temporary patch for https://github.com/cosmo-workspace/cosmo/issues/598
+ */
+
+package v1alpha1
+
+func (r *LoginRequest) GetId() string {
+	return r.Id
+}
+
+func (r *LoginRequest) GetPassword() string {
+	return r.Password
+}

--- a/hack/proxy-driver/main.go
+++ b/hack/proxy-driver/main.go
@@ -87,7 +87,7 @@ func main() {
 
 	srv.SetupSessionStore(authKey, encryptKey)
 
-	a := auth.NewHTTPAuthorizer(authURL, caCert)
+	a := auth.NewDashboardAuthorizer(authURL, caCert)
 	srv.SetupAuthorizer(a)
 
 	errChan := make(chan error)

--- a/internal/dashboard/auth_handler.go
+++ b/internal/dashboard/auth_handler.go
@@ -81,7 +81,7 @@ func (s *Server) Login(ctx context.Context, req dashv1alpha1.LoginRequest) (dash
 		log.Info("authrizer not found", "userid", req.Id, "authType", user.Spec.AuthType)
 		return ErrorResponse(log, kosmo.NewServiceUnavailableError("incorrect user or password", nil))
 	}
-	verified, err := authrizer.Authorize(ctx, req)
+	verified, err := authrizer.Authorize(ctx, &req)
 	if err != nil {
 		log.Error(err, "authorize failed", "userid", req.Id)
 		return ErrorResponse(log, kosmo.NewForbiddenError("incorrect user or password", nil))

--- a/pkg/auth/authorizer.go
+++ b/pkg/auth/authorizer.go
@@ -2,10 +2,13 @@ package auth
 
 import (
 	"context"
-
-	authv1alpha1 "github.com/cosmo-workspace/cosmo/proto/gen/auth-proxy/v1alpha1"
 )
 
 type Authorizer interface {
-	Authorize(ctx context.Context, msg *authv1alpha1.LoginRequest) (bool, error)
+	Authorize(ctx context.Context, msg AuthRequest) (bool, error)
+}
+
+type AuthRequest interface {
+	GetPassword() string
+	GetId() string
 }

--- a/pkg/auth/dashboard_authorizer.go
+++ b/pkg/auth/dashboard_authorizer.go
@@ -9,17 +9,17 @@ import (
 	"fmt"
 	"net/http"
 
-	authv1alpha1 "github.com/cosmo-workspace/cosmo/proto/gen/auth-proxy/v1alpha1"
+	dashv1alpha1 "github.com/cosmo-workspace/cosmo/api/openapi/dashboard/v1alpha1"
 )
 
-// HTTPAuthorizer authorize with cosmo-dashboard login API
-type HTTPAuthorizer struct {
+// DashboardAuthorizer authorize with cosmo-dashboard login API
+type DashboardAuthorizer struct {
 	URL    string
 	Client *http.Client
 }
 
-func NewHTTPAuthorizer(url string, ca []byte) *HTTPAuthorizer {
-	auth := &HTTPAuthorizer{
+func NewDashboardAuthorizer(url string, ca []byte) *DashboardAuthorizer {
+	auth := &DashboardAuthorizer{
 		URL: url,
 	}
 
@@ -36,9 +36,14 @@ func NewHTTPAuthorizer(url string, ca []byte) *HTTPAuthorizer {
 	return auth
 }
 
-func (a *HTTPAuthorizer) Authorize(ctx context.Context, msg *authv1alpha1.LoginRequest) (bool, error) {
+func (a *DashboardAuthorizer) Authorize(ctx context.Context, msg AuthRequest) (bool, error) {
 	body := new(bytes.Buffer)
-	err := json.NewEncoder(body).Encode(msg)
+
+	req := dashv1alpha1.LoginRequest{
+		Id:       msg.GetId(),
+		Password: msg.GetPassword(),
+	}
+	err := json.NewEncoder(body).Encode(req)
 	if err != nil {
 		return false, fmt.Errorf("failed to encode auth request: %w", err)
 	}

--- a/pkg/auth/password_secret_authorizer.go
+++ b/pkg/auth/password_secret_authorizer.go
@@ -4,7 +4,6 @@ import (
 	"context"
 
 	"github.com/cosmo-workspace/cosmo/pkg/auth/password"
-	authv1alpha1 "github.com/cosmo-workspace/cosmo/proto/gen/auth-proxy/v1alpha1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
@@ -17,7 +16,7 @@ func NewPasswordSecretAuthorizer(c client.Client) *PasswordSecretAuthorizer {
 	return &PasswordSecretAuthorizer{c}
 }
 
-func (a *PasswordSecretAuthorizer) Authorize(ctx context.Context, msg *authv1alpha1.LoginRequest) (bool, error) {
-	verified, _, err := password.VerifyPassword(ctx, a.Client, msg.Id, []byte(msg.Password))
+func (a *PasswordSecretAuthorizer) Authorize(ctx context.Context, msg AuthRequest) (bool, error) {
+	verified, _, err := password.VerifyPassword(ctx, a.Client, msg.GetId(), []byte(msg.GetPassword()))
 	return verified, err
 }

